jwst/resample/resample_spec_step.py

```diff
@@ -83,6 +82,17 @@ def process(self, input):
 
         # Call resampling
         self.drizpars = kwargs
+
+        #  If input is a 3D rateints (which is unsupported) skip the step
+        try:
+            if len((input_models[0][0]).shape) == 3:
```

Comment:
I am trying to understand this logic. `input_models`, according to the docs, can be a `MultiSlitModel`, a `SlitModel`, an`ImageModel`, or a `ModelContainer`. In all cases it is turned into a `ModelContainer` by the time it reaches this logic. The `ModelContainer` can have models which are "single" models, like `SlitModel` or models which are instances of `MultiSlitModel`. 
I think the logic here works correctly in the case when the input is a `MultiSlitModel`. Otherwise it raises an `AssetionError` but this error is raised for unrelated reasons. Essentially it is raised in the code which flattens the `meta` and allows access using  `.` (dots), as in `model.meta....` and has nothing to do with the data shape. I think better logic is necessary.

Reply:
For the rateints check for NIRSpec fixed slits, I updated the code to also check if the input is a MultiSlitModel.

Reply:
I don't think this was the right solution. 

The goal is to check the shape of the data array. The current implementation checks that an `AssertionError` is raised as a sign that the data array is 3D and this is incorrect. This `AssertionError` is raised in the base `DataModel` class, in the function that allows `.` dot access to the `meta` attribute (for example, instead of typing `model['meta'].instance['instrument']['name']` one can type `model.meta.instrument.name`. ) The code in the base class that does this uses `__getitem__` and expects the key to be of type string, not integer as is the typical case when indexing. Internally the function uses `assert`, hence the `AssertionError`, and it has nothing to do with the shape of the data array. So a `try ... except` block that catches an `AssertionError` is not an evidence that the array is 3D.

There are various alternative solutions, the simplest one may be to switch the order of checks in the condition from
```
if (len((input_models[0][0]).shape) == 3) and model_is_msm:
```
to
```
if model_is_msm and len((input_models[0][0]).shape) == 3:
```
the difference being that if it's not a `MultiSlit` model it doesn't reach the code that raises `AssertionError`. 

That said, this check should be done in the beginning of the function, immediately after opening the input and then it will be much simpler and faster because it won't have to package the input data into a `ModelContainer`. A `MultiSlit` model is created in `extract_2d`. It's either a grism observation or a Nirspec observation, excluding BOTS mode (which creates a single `SlitModel`). Since grism observations are not resampled I think it's OK to check that the input is a `MultiSlitModel` immediately after the input is opened (as you do now on line 33) and if so check that the data array of the first slit is 3D. I think that would be the cleanest solution.